cytotable/utils.py

```diff
@@ -75,3 +79,38 @@ def _duckdb_with_sqlite() -> duckdb.DuckDBPyConnection:
         LOAD sqlite_scanner;
         """
     )
+
+
+def _cache_cloudpath_to_local(path: Union[str, AnyPath]) -> pathlib.Path:
+    """
+    Takes a cloudpath and uses cache to convert to a local copy
+    for use in scenarios where remote work is not possible (sqlite).
+
+    Args:
+        path: Union[str, AnyPath]
+            A filepath which will be checked and potentially
+            converted to a local filepath.
+
+    Returns:
+        pathlib.Path
+            A local pathlib.Path to cached version of cloudpath file.
+    """
+    if (
+        # check that the filepath includes cloud provider prefix
+        any(
+            str(path).lower().startswith(cloudtype)
+            for cloudtype in ["s3://", "gc://", "az://"]
+        )
+        # check that the path is a file (caching won't work with a dir)
+        and AnyPath(path).is_file()
+        # check that the file is of sqlite type
+        # (other file types will be handled remotely in cloud)
+        and AnyPath(path).suffix.lower() == ".sqlite"
+    ):
+        # incur a read which will trigger caching of the file
+        CloudPath(path).read_bytes()
+        # update the path to be the local filepath for reference in CytoTable ops
+        path = CloudPath(path).fspath
+
+    # cast the result as a pathlib.Path
+    return pathlib.Path(path)
```

Comment:
From the docs, seems that just referencing `.fspath` is enough to cause the remote file to be cached, which I imagine would make it unnecessary to call `read_bytes()` on it: https://cloudpathlib.drivendata.org/stable/caching/#accessing-the-cached-version-directly-read-only.

Also, I could imagine `clouthpathlib` being extended with additional providers in the future. If that seems like something you'd want, you might defer to the library to determine if it supports the given URL, e.g.:

```python
candidate_path = AnyPath(path)
  
if candidate_path.is_file() and candidate_path.suffix.lower() == ".sqlite":
  try:
    path = CloudPath(path).fspath
  except clouthpathlib.exceptions.CloudPathException:
    # you might consider logging here, up to you
    pass
    
return pathlib.Path(path)
```

Reply:
Thanks for this, great thoughts! I wrestled with both of these a bit and welcome the chance to improve.
- `fspath` vs `read_bytes`: great point, I'll get this changed. I don't feel strongly about it but wondering aloud if `read_bytes` is less presumptive about the underlying operations (perhaps a cloudpath could be understood to be a type of "filesystem" in some circumstances).
- additional providers: One reason for the explicit checks was that I couldn't get [`CloudPath.is_valid_cloudpath`](https://cloudpathlib.drivendata.org/v0.6/api-reference/cloudpath/#cloudpathlib.cloudpath.CloudPath.is_valid_cloudpath) to work in expected ways. I like the alternative of a try/except here better for the reasons you mentioned. I'll these a try in an update.